data/collector.go

```diff
@@ -59,6 +59,7 @@ type collector struct {
 	cancelCtx         context.Context
 	cancel            context.CancelFunc
 	capturer          Capturer
+	closed            bool
```

Comment:
It's being used in the Closed() to prevent it from panicking if already closed. Originally was thinking that a collector might close, but not be deleted. But then I added the delete because a new collector should be generated and a new file path should, so maybe no longer needed. will review

Reply:
Which line in close was panicking before? Adding this doesn't seem unreasonable to me and it's probably better to be defensive in this case re panics, moreso asking if the var is necessary to prevent that